Don't build Linux-specific SELINUX extension support on Solaris

Upstream always tries to generate the source files even if they
won't be built, but we don't have the xselinux.xml needed to do that.

--- libxcb-1.14/src/Makefile.am
+++ libxcb-1.14/src/Makefile.am
@@ -204,8 +204,8 @@ libxcb_xprint_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xprint_la_SOURCES = xprint.c xprint.h
 endif
 
-EXTSOURCES += xselinux.c
 if BUILD_SELINUX
+EXTSOURCES += xselinux.c
 lib_LTLIBRARIES += libxcb-xselinux.la
 libxcb_xselinux_la_LDFLAGS = -version-info 0:0:0 -no-undefined @lt_enable_auto_import@
 libxcb_xselinux_la_LIBADD = $(XCB_LIBS)
